bot.js

```diff
@@ -335,36 +343,82 @@ bot.isChannel = function(name) {
 };
 
 
-bot.getReply = function(chan) {
-  return function(args) {
+bot.getReply = function(to, isPm, pmTarget) {
+  var spamReply = function(args) {
     var repStr = bot.stringifyArgs.apply(this, arguments);
-    bot.client.say(chan, repStr);
+    bot.client.say(to, repStr);
   };
-};
 
-bot.getNoticeReply = function(to) {
-  return function(args) {
-    var repStr = bot.stringifyArgs.apply(this, arguments);
+  // custom reply takes options of the following:
+  // {
+  //   lines: 2, // number of lines to spam
+  //   replaceNewLines: false, // whether to replace newlines with | while figuring out if it'll spam
+  //   pmExtra: false, // whether to PM the whole message if it overflows
+  // }
+  var customReply = function(opts, args) {
+    var maxLineChars = 460; // convenient lie; the true number is complicated due to having to account for nick-length etc
 
-    if(bot.isChannel(chan)) {
-      bot.callModuleFn('channotice', [bot.client.nick, to, repStr]);
-    } else {
-      bot.callModuleFn('pmnotice', [bot.client.nick, to, repStr]);
+    args = Array.prototype.slice.call(arguments, 1);
+
+    // If it's a pm, all options get ignored
+    if(isPm) {
+      spamReply.apply(this, args);
+    }
+
+    var repStr = bot.stringifyArgs.apply(this, args);
```

Comment:
Nit: I'd add a comment here saying something to the effect of "calculate number of messages to send these lines"?